packages/web3-utils/src/converters.ts

```diff
@@ -411,6 +412,7 @@ export const toHex = (
  */
 export const toNumber = (value: Numbers): number | bigint => {
 	if (typeof value === 'number') {
+			console.warn('Warning: Using type `number` with values that are large or contain many decimals may cause loss of precision, it is recommended to use type `string` or `BigInt` when using conversion methods')
             if (value > 1e+20) {
```

Comment:
What about handling things here for small numbers as well?

Reply:
that makes sense to me 👍